front/src/common/Map/WarpedMap/core/helpers.ts

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable prefer-destructuring, no-plusplus */
+import { Feature, FeatureCollection, LineString, Point, Polygon, Position } from 'geojson';
+import _, { clamp, first, last, mapValues } from 'lodash';
+import length from '@turf/length';
+import { point } from '@turf/helpers';
+import along from '@turf/along';
+import distance from '@turf/distance';
+import { MapGeoJSONFeature } from 'maplibre-gl';
+
+import { EditoastType, LAYER_TO_EDITOAST_DICT, LayerType } from 'applications/editor/tools/types';
+import { getMixedEntities } from 'applications/editor/data/api';
+import { flattenEntity } from 'applications/editor/data/utils';
+import vec, { Vec2 } from 'common/Map/WarpedMap/core/vec-lib';
+
+/**
+ * Useful types:
+ */
+export type TriangleProperties = { triangleId: string };
+export type Triangle = Feature<Polygon, TriangleProperties>;
+export type BarycentricCoordinates = [number, number, number];
+export type GridFeature = FeatureCollection<Polygon, TriangleProperties>;
+export type GridIndex = Record<string, Triangle>;
+export type Grids = {
+  regular: GridFeature;
+  warped: GridFeature;
+};
+export type PointsGrid = Record<number, Position>[];
+
+/**
+ * Path manipulation helpers:
+ */
+export function getSamples(
+  line: Feature<LineString>,
+  samples: number
+): { step: number; points: Feature<Point>[] } {
+  if (samples <= 1) throw new Error('samples must be an integer greater than 1');
+
+  const points: Feature<Point>[] = [];
+  const l = length(line, { units: 'meters' });
+  const step = l / (samples - 1);
+  for (let i = 0; i < samples; i++) {
+    if (!i) {
+      points.push(point(first(line.geometry.coordinates) as Position));
+    } else if (i === samples - 1) {
+      points.push(point(last(line.geometry.coordinates) as Position));
+    } else {
+      const at = clamp(step * i, 0, l);
```

Comment:
I think the clamping is not really required here ? (but you can keep it for corner cases I imagine)
```suggestion
      const at = step * i;
```

Reply:
Right, it seems useless indeed, I'll remove it.